starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -2269,7 +2269,7 @@ sub print_report {
 	my $bad_FOM = $self->{figure_of_merit}->{cum_prob_bad};
 	$o .= "$red_font_start" if $bad_FOM;
 	$o .= "Probability of acquiring 2 or fewer stars (10^-x):\t";
-        $o .= substr(sprintf("%.4f", $self->{figure_of_merit}->{P2}), 0, 6) . "\t";
+        $o .= substr(sprintf("%.1f", $self->{figure_of_merit}->{P2}), 0, 3) . "\t";
```

Comment:
What is the `substr` for here? I see the logic in the change but the original code is confusing me.

Reply:
I am betting this can be removed.

I can imagine something like this to truncate the number if the rounded number takes more digits than expected, but it is also not a very good idea.

Reply:
this could related to sprintf actually rounding a float and not giving a string as output. When you round, you might not get a number with exact binary representation, and then when you print you get 0.9000001 instead of 0.9, so you truncate the string.

Reply:
As long as I'm trying out perl... I feel like the `substr` is not helping out here. In the unlikely event that P2 is > 10 or negative, the `substr` would give a bad result.
```
perl -e '$a="10.23432"; $b=substr($a, 0, 3); print("$b\n")'
10.
```